modules/platforms/dotnet/Apache.Ignite/Internal/Compute/Compute.cs

```diff
@@ -69,7 +69,7 @@ public Compute(ClientFailoverSocket socket, Tables tables)
             IgniteArgumentCheck.NotNull(nodes);
             IgniteArgumentCheck.NotNull(jobClassName);
 
-            return await ExecuteOnOneNode<T>(GetRandomNode(nodes), units, jobClassName, args).ConfigureAwait(false);
+            return await ExecuteOnNodes<T>(GetRandomNode(nodes), nodes, units, jobClassName, args).ConfigureAwait(false);
```

Comment:
[Multiple enumeration of IEnumerable](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1851).

Let's do the following:
* Change `GetRandomNode` and `ExecuteOnNodes` to require `ICollection<IClusterNode>`
* Convert `IEnumerable<IClusterNode>` to `ICollection<IClusterNode>` with `GetNodesCollection` once on the previous line